src/python_pybind11/CMakeLists.txt

```diff
@@ -1,3 +1,19 @@
+# Detect if we are doing a standalone build of the bindings, using an external gz-math
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  cmake_minimum_required(VERSION 3.16)
+  set(GZ_MATH_VER 8)
+  project(gz-math${GZ_MATH_VER}-python VERSION ${GZ_MATH_VER})
+  find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
```

Comment:
I've moved the `find_package(pybind11)` call here in https://github.com/gazebosim/gz-math/pull/636/commits/06b8ab8b275650309f45578d9fe852d64dade47a using the `CMAKE_REQUIRE_FIND_PACKAGE_pybind11` variable to support finding it as an optional or required package depending on the context.

It's tricky to move the `find_package(Python3)` call to this folder because the `Development` component can be optional or required, depending on the context, but the `FindPython3` module doesn't support `CMAKE_REQUIRE_FIND_PACKAGE_*` for its components. So I just left the `find_package(Python3)` code as is